pkg/kubernetes/job_test.go

```diff
@@ -84,5 +85,55 @@ var _ = Describe("Job", func() {
 	})
 
 	Describe("#Status", func() {
+		var s manifest.Status
+
+		BeforeEach(func() {
+			o := job.Object()
+			completions := int32(1)
+			o.Status.Succeeded = 1
+			o.Spec.Completions = &completions
+		})
+
+		JustBeforeEach(func() {
+			s = job.Status()
+		})
+
+		Context("succeeded pods is less than completions", func() {
+			BeforeEach(func() {
+				o := job.Object()
+				completions := int32(2)
+				o.Status.Succeeded = 1
+				o.Spec.Completions = &completions
+			})
+
+			When("there is a failed condition", func() {
+				BeforeEach(func() {
+					o := job.Object()
+					o.Status.Conditions = []v1.JobCondition{
+						{
+							Type: v1.JobFailed,
+						},
+					}
+				})
+
+				It("returns status failed", func() {
+					Expect(s.Failed.State).To(BeTrue())
+				})
+			})
+
+			When("the job is not finished", func() {
+				It("returns the expected status", func() {
+					Expect(s.Stable.State).To(BeFalse())
+					Expect(s.Stable.Message).To(Equal("Waiting for jobs to finish"))
+				})
+			})
+		})
+
+		When("it succeeeds", func() {
```

Comment:
I found this confusing because it's at the same level as the context block and the description is ambiguous.

Reply:
This is just the success case. I typically refer to the happy path as "when it succeeds".